modules/images/image-loading-optimization/detection.php

```diff
@@ -35,6 +35,9 @@ function ilo_get_detection_script( string $slug, array $needed_minimum_viewport_
 	 */
 	$detection_time_window = apply_filters( 'ilo_detection_time_window', 5000 );
 
+	$web_vitals_version = json_decode( file_get_contents( __DIR__ . '/../../../package.json' ), true )['dependencies']['web-vitals']; // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local file, not a remote URL.
+	$web_vitals_lib_src = add_query_arg( 'ver', $web_vitals_version, plugin_dir_url( __FILE__ ) . '/web-vitals/web-vitals.js' );
```

Comment:
This is loading the non-module version, so adding a `ver` will work fine.

Nevertheless, for the development version of the plugin should we not load the file directly from `node_modules`? Once this is built for distribution, there won't be a `package.json` either, so we wouldn't be able to read from it. So perhaps for development we should have the following:

```suggestion
	$web_vitals_version = json_decode( file_get_contents( __DIR__ . '/../../../package.json' ), true )['dependencies']['web-vitals']; // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local file, not a remote URL.
	$web_vitals_lib_src = add_query_arg( 'ver', $web_vitals_version, plugin_dir_url( __FILE__ ) . '/../../../../node_modules/web-vitals/web-vitals.js' );
```

Then as part of the plugin build process, these get rewritten as after `copy-vendor-libs` is run?

I think there should be some plugin-specific logic here:

https://github.com/WordPress/performance/blob/fd23971345680bf13d8e697f68bca5ad92eb855c/bin/plugin/commands/build-plugins.js#L67-L94

Reply:
Or not per https://github.com/WordPress/performance/pull/953#discussion_r1464035683

But still there is the question of where the package version would be obtained if the `package.json` isn't present.